provider_test.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestValidateFilePath(t *testing.T) {
+	// Don't use filepath.Join to generate the test cases because it calls filepath.Clean
+	// which simplifies some of the test cases into less interesting paths.
+	for _, tc := range []string {
```

Comment:
I wonder if environment variables (bash expansion) is also an attack vector we need to consider? I don't think so but calling it out. I'm just worried about any other bash tricks that could cause problems here.

Reply:
That's a good thought. I think you're correct that it's not a consideration, because
a) the code that actually writes the file is within this provider, so we don't have to worry about expansion beyond the confines of this code base
b) `ioutil.WriteFile` doesn't do anything funky. It goes straight to `os.Open` which uses a syscall to open the file handle, so there shouldn't be an env expansion in there.